### ISSUE TITLE
CRAN packages with -Wpointer-sign warnings External

### ISSUE THREAD

**Issue by mbjones** [ropensci/redland-bindings#96]
Email from CRAN seems to ask us to fix the warnings in the underlying C package:

> Seen with both gcc and clang.  That is
> C50 Cubist RMVL RecordLinkage Rttf2pt1 RxODE blockTools bnstruct not
randaes redland
> Please correct before 2022-01-23 to safely retain your package on CRAN.
>
>-- 
>Brian D. Ripley,                  ripley@stats.ox.ac.uk
>Emeritus Professor of Applied Statistics, University of Oxford

@gothub let's discuss please in terms of what this would take.

**Comment by gothub** [comment]
The warnings that were reported by CRAN are the same ones that have been with the package for the last few years.

They result from the SWIG generated `redland.c` file, which contains an incorrect function declaration, for example:
```
  librdf_digest_update(arg1,(char const *)arg2,arg3);
```

instead of 
```
  librdf_digest_update(arg1,(unsigned char const *)arg2,arg3);
```

For a quick fix that will allow a CRAN resubmission before 20220123, I have manually updated `redland.c` so that the package builds cleanly, i.e. `CMD INSTALL --no-multiarch --with-keep.source redland` produces no warnings.

With these changes to `redland.c`, `devtools:check()` and `devtools:test()` pass without errors or warnings.

I'll file an issue with https://github.com/swig/swig/issues after creating a simple test file that they can use for debugging.

Note that this compile problem occurs with the most recent SWIG version (SWIG 4.0.2, 2020/06/08), with the unmodified `redlands.c` file.

I'll also test package build/install on other platforms via rhub and post the results here.

**Comment by gothub** [comment]
rhub builds:
- Fedora Linux, R-devel, clang, gfortran: OK
- Fedora Linux, R-devel, GCC: OK
- Debian Linux, R-release, GCC: OK
- Windows Server 2008 R2 SP1, R-release, 32/64 bit: NOTE
  installed size is  5.9Mb
  sub-directories of 1Mb or more:
    libs   5.0Mb

**Comment by cboettig** [comment]
:clap: heroic work, Peter!

**Comment by gothub** [comment]
@mbjones the updated `redlands.c` file is on the `feature-R-compile-errors-#96` branch. Should this be merged now and a new release submitted with just that change?

**Comment by mbjones** [comment]
I didn't test it, but if all the tests are passing, then yeah, let's get it in a release. I did ask about whether you wanted to fold that longstanding PR in as well, but that requires a doc rebuild. Your call there. Why don't you merge your branch into develop and we'll prep the release there as usual.

**Comment by gothub** [comment]
Fixed in commit 2018aeb178ba88e6c5ca40b9ce47b01d7da89c1c

Updated `./src/redland.c` directly to fix the compile errors.
